src/Features/CSharp/Portable/EditAndContinue/SyntaxUtilities.cs

```diff
@@ -116,9 +116,25 @@ public static void AssertIsBody(SyntaxNode syntax, bool allowLambda)
                 return;
             }
 
+            // special case for top level statements, which have no containing block other than the compilation unit
+            if (syntax is CompilationUnitSyntax unit && unit.ContainsTopLevelStatements())
+            {
+                return;
+            }
+
             Debug.Assert(false);
         }
 
+        public static bool ContainsTopLevelStatements(this CompilationUnitSyntax compilationUnit)
+        {
+            if (compilationUnit.Members.Count == 0)
+            {
+                return false;
+            }
+
+            return compilationUnit.Members[0] is GlobalStatementSyntax;
```

Comment:
It seems like the callers often do this same access. Maybe it would make sense to overload this to return the first top level statement as an `out` parameter, or to just change the signature and have callers who don't care about the actual statement use `out _`.

Reply:
I'm going to leave this for now - 2 callers don't access Members again, 2 callers use the first and last elements - but will keep it in mind if there is more usage. It feels odd to only return the first node when the callers use the first and last, but finding the last node seems wasteful if 50% of the time its not needed.